units/crons.go

```diff
@@ -68,7 +68,10 @@ func StartCrons(ctx context.Context, env cedar.Environment, rpcTLS bool) error {
 			return nil
 		}
 
-		return queue.Put(ctx, NewRemoteAmboyStatsCollector(env, utility.RoundPartOfMinute(0).Format(tsFormat)))
+		catcher := grip.NewBasicCatcher()
+		catcher.Add(queue.Put(ctx, NewRemoteAmboyStatsCollector(env, utility.RoundPartOfMinute(0).Format(tsFormat))))
+		catcher.Add(queue.Put(ctx, NewStatsCacheLogger(env, utility.RoundPartOfMinute(0).Format(tsFormat))))
```

Comment:
The issue with creating this as an amboy job is that any of the 12 servers can pick it up. So it becomes very easy where we run into a case where these stats are not being flushed frequently enough on each server and we start dropping many useful stats. I would propose instead of using an amboy job to create a background thread. We could also then just move the stats collection entirely into the `rpc` package (since it is specific to that). What are your thoughts?

Reply:
That's a good point. I moved it into a loop that gets started together with the consumer loop.
I like the location of the stats cache itself: I was aiming to consolidate global state to just the environment. I'm agnostic as to adding the stats from the rpc service vs. the model.